webhooks.py

```diff
@@ -132,12 +133,19 @@ def index():
     # so let's be safe
     name = payload['repository']['name'] if 'repository' in payload else None
 
+
     meta = {
         'name': name,
         'branch': branch,
         'event': event
     }
-    logging.info('Metadata:\n{}'.format(dumps(meta)))
+
+    if event == 'push' and payload['deleted']:
+        logging.info("Skipping push for deletion for %s" % meta)
+        return "Skipped"
+
```

Comment:
Yes - so I noticed when branch deletion is made in git code, two things happen:
- you receive action "delete" with branch name
- you receive "push" with state "deleted" 

this causes that hooks for handling push and deletetion - what is even worse - they can be called in random order. I found this annoying and hard to use, when I have some project build script put inside push-hook , and project deletion inside delete-hook, then project can be rebuild AFTER it is deleted and so on. 

In my opinion this is simpler and give some abstractino over hook scripts that they will be run only once on particular action that happen.